fixtures/futures/tests/bindings/test_futures.swift

```diff
@@ -232,4 +232,33 @@ Task {
 	counter.leave()
 }
 
+// Test a future that uses a lock and that is cancelled.
+counter.enter()
+Task {
+	let task = Task {
+	    try! await useSharedResource(options: SharedResourceOptions(releaseAfterMs: 100, timeoutMs: 1000))
+	}
+
+	// Wait some time to ensure the task has locked the shared resource
+	try await Task.sleep(nanoseconds: 50_000_000)
+	// Cancel the job task the shared resource has been released.
+        //
+        // FIXME: this test currently passes because `test.cancel()` doesn't actually cancel the
+        // operation.  We need to rework the Swift async handling to handle this properly.
```

Comment:
```suggestion
    //
    // FIXME: this test currently passes because `test.cancel()` doesn't actually cancel the
    // operation.  We need to rework the Swift async handling to handle this properly.
```